apps/webservice/src/app/api/v1/config/route.ts

```diff
@@ -0,0 +1,299 @@
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+import jsYaml from "js-yaml";
+import _ from "lodash";
+import { isPresent } from "ts-is-present";
+
+import { can } from "@ctrlplane/auth/utils";
+import { and, eq, inArray, sql, takeFirstOrNull } from "@ctrlplane/db";
+import { db } from "@ctrlplane/db/client";
+import * as schema from "@ctrlplane/db/schema";
+import {
+  cancelOldReleaseJobTriggersOnJobDispatch,
+  createJobApprovals,
+  createReleaseJobTriggers,
+  dispatchReleaseJobTriggers,
+  isPassingAllPolicies,
+  isPassingReleaseStringCheckPolicy,
+} from "@ctrlplane/job-dispatch";
+import { Permission } from "@ctrlplane/validators/auth";
+import { cacV1 } from "@ctrlplane/validators/cac";
+
+import { getUser } from "~/app/api/v1/auth";
+
+const { entries } = Object;
+
+const upsertSystems = async (db: Tx, config: CacV1) => {
+  if (config.systems == null || entries(config.systems).length == 0) return;
+
+  const systemInserts = entries(config.systems ?? {}).map(([slug, system]) => ({
+    name: system.name ?? slug,
+    description: system.description ?? "",
+    workspaceId: config.workspace,
+    slug,
+  }));
+
+  if (systemInserts.length == 0) return;
+
+  await db
+    .insert(schema.system)
+    .values(systemInserts)
+    .onConflictDoUpdate({
+      target: [schema.system.workspaceId, schema.system.slug],
+      set: {
+        name: sql`excluded.name`,
+        description: sql`excluded.description`,
+      },
+    });
+};
+
+const upsertDeployments = async (db: Tx, config: CacV1) => {
+  if (config.deployments == null || entries(config.deployments).length == 0)
+    return;
+
+  const deploymentInfo = entries(config.deployments ?? {})
+    .map(([slug, deployment]) => {
+      const [systemSlug, deploymentSlug] = slug.split("/");
+      if (systemSlug == null || deploymentSlug == null) return null;
+
+      return {
+        systemSlug,
+        deployment: {
+          name: deployment.name ?? deploymentSlug,
+          slug: deploymentSlug,
+          description: deployment.description ?? "",
+          jobAgentId: deployment.jobAgent?.id,
+          jobAgentConfig: deployment.jobAgent?.config ?? {},
+        },
+      };
+    })
+    .filter(isPresent);
+  const systemSlugs = deploymentInfo.map(({ systemSlug }) => systemSlug);
+
+  const systems = await db
+    .select()
+    .from(schema.system)
+    .where(inArray(schema.system.slug, systemSlugs));
+
+  const jobAgentsIds = await db
+    .select({ id: schema.jobAgent.id })
+    .from(schema.jobAgent)
+    .where(
+      and(
+        inArray(
+          schema.jobAgent.id,
+          deploymentInfo
+            .map(({ deployment }) => deployment.jobAgentId)
+            .filter(isPresent),
+        ),
+        eq(schema.jobAgent.workspaceId, config.workspace),
+      ),
+    )
+    .then((rows) => rows.map(({ id }) => id));
+
+  const systemMap = new Map(systems.map((system) => [system.slug, system]));
```

Comment:
```suggestion
  const systemMap = Object.fromEntires(systems.map((system) => [system.slug, system]));
```

Is there a specific reason for maps?